src/CMetricsCalculator.cpp

```diff
@@ -300,7 +318,13 @@ CMetricsCalculator::calculate_metrics_switch()
 			} else
 				STYLE_HINT(NO_SPACE_AFTER_SEMICOLON);
 		}
+		in_struct = false;
+		saw_struct = false;
+		in_union = false;
+		saw_union = false;
 		bol.saw_non_space();
+        /* if (!in_function) */
```

Comment:
What's with the commented code?  If it's a comment rephrase, if it's commented out code, please remove.